Mathlib/Topology/Basic.lean

```diff
@@ -519,6 +519,13 @@ theorem closure_iUnion_of_finite [Finite ι] (f : ι → Set X) :
   rw [← sUnion_range, (finite_range _).closure_sUnion, biUnion_range]
 #align closure_Union closure_iUnion_of_finite
 
+@[simp]
+theorem closure_iUnion₂_of_finite {n : ℕ} (f : ℕ → Set X) :
+    closure (⋃ m ≤ n, f m) = ⋃ m ≤ n, closure (f m) := by calc
+  closure (⋃ m ≤ n, f m) = closure (⋃ m ∈ {m | m ≤ n}, f m) := by simp only [mem_setOf_eq]
+  _ = ⋃ m ∈ {m | m ≤ n}, closure (f m) := Set.Finite.closure_biUnion (finite_le_nat n) f
+  _ = ⋃ m ≤ n, closure (f m) := by simp only [mem_setOf_eq]
```

Comment:
Again this should be true by definition. If this is true then the whole proof should be `:= (finite_le_nat n).closure_biUnion f`, without entering tactic mode.

Reply:
870a04d